src/openeo_processes/cubes.py

```diff
@@ -2066,15 +2066,18 @@ def exec_xar(predictors, target, training, num_trees = 100, max_variables = None
             target = gpd.GeoDataFrame.from_file(target)
 
         if isinstance(predictors, gpd.geodataframe.GeoDataFrame) and isinstance(target, gpd.geodataframe.GeoDataFrame):
-            predictors_pandas = pd.DataFrame(predictors)
-            predictors_pandas = predictors_pandas.drop(columns=['geometry'])
+            if (predictors['geometry'] == target['geometry']).all():
+                predictors_pandas = pd.DataFrame(predictors)
+                predictors_pandas = predictors_pandas.drop(columns=['geometry'])
+                target_pandas = pd.DataFrame(target)
+                target_pandas = target_pandas.drop(columns=['geometry'])
+            else:
+                raise Exception('Geometries of input predictors and target do not match.')
```

Comment:
```suggestion
            if not (predictors['geometry'].equals(target['geometry']):
                raise Exception('Geometries of input predictors and target do not match.')
                
            predictors_pandas = pd.DataFrame(predictors)
            predictors_pandas = predictors_pandas.drop(columns=['geometry'])
            target_pandas = pd.DataFrame(target)
            target_pandas = target_pandas.drop(columns=['geometry'])
```

I'd rather use the inbuilt functions for checking equality!
Also, if we raise the error right away, we don't have to obscure this logic in a conditional!